src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1088,6 +1119,10 @@ public InMemoryOutput downloadOutputs(RemoteAction action, RemoteActionResult re
     tmpOutErr.clearOut();
     tmpOutErr.clearErr();
 
+    if (!forcedDownloads.isEmpty()) {
+      moveOutputsToFinalLocation(forcedDownloads);
```

Comment:
Move this under the `else` case. That also means we don't have to distinguish between `downloads` and `forcedDownloads`. (Also, if `moveOutputsToFinalLocation` can handle an empty list, then call it unconditionally.)

Reply:
This may change the handling of outerr, in the `!downloadOutputs` case, which seems ok to me, but it's a semantic change.

Reply:
I tried a few versions of this code and I'm not too comfortable with moving this to the else case. Not sure the readability benefit of unifying downloads and forcedDownloads is worth the risk of potentially breaking the other modes of operation. If you're ok with this I'd rather leave it as is. I've simplified the rest of the code a lot now.

Reply:
I tried a few versions of this code and I'm not too comfortable with moving this to the else case. Not sure the readability benefit of unifying downloads and forcedDownloads is worth the risk of potentially breaking the other modes of operation. If you're ok with this I'd rather leave it as is. I've simplified the rest of the code a lot now.